### DISCUSSION TITLE
Playtime2: Recording the processed sound (amp sim) into the clip?

### DISCUSSION THREAD

**Discussion by Chianti** [helgoboss/helgobox#1324] [Q&A] [answered]
Can anyone help me understand how to get the following workflow (in Reaper):

When I record into a Playtime slot, I want the guitar sim I'm using on the track _to be recorded on the clip_ (post FX).
For the next slot, I want to change the amp sim patch, and record its different sound into the next clip.
And so on.

What's happening just now is that only the dry direct input is being recorded (pre FX). Even if I put the amp sim on the InFX tray, the clip is still recorded clean (which really confuses me, because the first thing the signal is supposed to hit upon entering Reaper is the InFX - Playtime is intercepting the signal before even that?).

Thanks.

**Comment by helgoboss** [comment]
Hi. For this you currently need to use REAPER's built-in audio loopback devices. Another user had a similar question recently and someone described the procedure: https://github.com/helgoboss/helgobox/discussions/1284#discussioncomment-11113030

**Comment by Chianti** [comment, marked as answer]
Hmm, that guy is recording onto the Reaper timeline onto different tracks than the input track, and doesn't seem to be recording into Playtime2 slots. Whereas I'm just trying to record a wet signal into the Playtime slots. I don't think that post is applicable?

I'm having a hard time understanding why having my amp sim on the **In**-FX tray isn't working? Does it mean that Playtime2 is placed before the In-FX tray in the incoming signal path, and is 'intercepting' the signal before it reaches the amp sim?

**EDIT: I got it working using the loopback. Steps are:
· Reaper Prefs > Audio > Virtual Loopback audio hardware channels = 2
· Create 'Guitar In' track. Monitor on. Record disabled. Amp sim in FX tray.
· Route > add new hardware output > 'hardware output loop output 1'
· Create 'Guitar1' track in Playtime2 (also creates Reaper track and links them)
· Record arm, but disable monitor. In PT2, choose 'loopback 1' as input (also changes Reaper track).
· In PT2, choose a slot in the 'Guitar 1' column and start recording into the PT2 slot.
  It will record the wet FX signal.**

I don't really understand why using loopback and another track is necessary for this though. In my head, the concept is that the signal goes into Reaper, through the track, through the FX tray, and gets recorded wet into a PT2 slot as a clip. I imagined that one of the main uses of PT2 straight out of the gate by default would be to record lots of different sounding clips in lots of slots, to test out different sounds and FX?

**Reply by helgoboss** [reply]
Ah sorry, you use input FX, I understand. Nope, this will be ignored. Playtime captures hardware inputs directly.

The workaround is still similar as the one in the other issue: Use audio loopback devices, instead of input FX.

If you are interested in input FX support, please create a feature request. Not sure if it's possible to support that, but having the FR is a start.

**Reply by Chianti** [reply]
> If you are interested in input FX support, please create a feature request.

Thanks but it's ok. It was just a bit weird having to set-up loopback and a second channel to do what I thought would be a default workflow. That's just me though - I'm not quite understanding all of PT2 conceptually yet.

**Reply by helgoboss** [reply]
Would be cool if input FX would be respected, I would love it to be part of the default workflow. I've thought about it a bit and I see a way this could be done. It's not super easy and I'm not sure yet if it would be solid/reliable. Let's see what the future offers.

**Reply by Chianti** [reply]
Since there is a working solution, I would imagine this is low priority though, especially if its programmatically difficult. Don't feel any pressure to work on it - it should feel like a fun challenge to do when you feel like it. As I said before, my main problem is my current lack of conceptual understanding of PT2, not with your software or the way you designed it! :)
